vmngclient/endpoints/__init__.py

```diff
@@ -479,8 +479,10 @@ def wrapper(*args, **kwargs):
             _kwargs = self.merge_args(args, kwargs)
             payload = _kwargs.get("payload")
             params = _kwargs.get("params")
-            self.url = self.url.format(**_kwargs)
-            response = _self._request(self.http_method, self.url, payload=payload, params=params, **self.kwargs)
+            self.formatted_url = self.url.format(**_kwargs)
```

Comment:
Better to use it as local variable. Formatted url could be different for each time endpoint is called, so it is not a property of endpoint (no need to store it's state - just prepare before sending)